static/js/typeahead_helper.js

```diff
@@ -79,14 +79,28 @@ var rendered = { persons: new Dict(), streams: new Dict(), user_groups: new Dict
 
 exports.render_person = function (person) {
     if (person.special_item_text) {
-        return exports.render_typeahead_item({ primary: person.special_item_text });
+        return exports.render_typeahead_item({
+            primary: person.special_item_text,
+            is_person: true,
+        });
     }
 
     var html = rendered.persons.get(person.user_id);
     if (html === undefined) {
+        var avatar_url = person.avatar_url;
+
+        if (!avatar_url) {
+            var hash = md5(person.email);
+            avatar_url = 'https://secure.gravatar.com/avatar/' + hash + '?d=identicon';
+        }
+
+        var small_avatar_url = people.format_small_avatar_url(avatar_url)
```

Comment:
I think we want all the logic around `gravatar` and things to live in `people.js` -- can you move this to a function there?  Might require some refactoring to avoid code duplication.

Reply:
@timabbott OK, I moved it to a separate function `small_avatar_url_for_person` and did some refactoring so that `small_avatar_url` uses that function as well if necessary. Thanks!